lib/actions/calendar.ts

```
@@ -48,51 +53,182 @@ export async function getNotes(date: Date, chatId: string | null): Promise<Calen
+    console.error("Error validating mentions:", error);
+    return [];
+  }
+}
+
+function sanitizePromptInput(content: string) {
+  // Robustly escape or remove dangerous characters to prevent prompt injection
+  return content
+    .replace(/\\/g, '\\\\') // Escape backslashes
+    .replace(/\$/g, '\\$')  // Escape dollar signs
+    .replace(/['"<>{}[]]/g, '') // Remove single quotes, angle brackets, curly braces, and square brackets
+    .replace(/[\x00-\x1F\x7F]/g, ' ') // Remove non-printable/control characters
```

Biome (2.3.13)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)